plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -120,29 +124,20 @@ mkTxOutput = \case
     PubKeyTarget pk vl -> pubKeyTxOut vl pk
     ScriptTarget hsh ds vl -> scriptTxOut' vl (Ledger.AddressOf (Scripts.unsafePlcAddress hsh)) ds
 
-PlutusTx.makeLift ''EscrowParams
 
 data Action = Redeem | Refund
 
-instance PlutusTx.IsData Action where
-    toData Redeem = PlutusTx.Constr 0 []
-    toData Refund = PlutusTx.Constr 1 []
-    {-# INLINABLE fromData #-}
-    fromData (PlutusTx.Constr i [])
-        | i == 0 = Just Redeem
-        | i == 1 = Just Refund
-    fromData _ = Nothing
-
+PlutusTx.makeIsData ''Action
 PlutusTx.makeLift ''Action
 
 {-# INLINABLE meetsTarget #-}
 meetsTarget :: PendingTx -> EscrowTarget -> Bool
 meetsTarget ptx = \case
     PubKeyTarget pk vl ->
-        valuePaidTo ptx pk == vl
+        valuePaidTo ptx pk `geq` vl
```

Comment:
This should probably have a note explaining the poisoning scenario and how this defends against it. I'd also love a test for that case!

Reply:
Done.